crates/pop-cli/src/commands/up/rollup.rs

```diff
@@ -198,36 +287,48 @@ fn prepare_reserve_call_data(chain: &Chain, proxy: &Proxy, cli: &mut impl Cli) -
 
 // Generates chain spec files for the project.
 async fn generate_spec_files(
+	deployment_config: &mut Deployment,
 	id: u32,
 	path: Option<&Path>,
 	cli: &mut impl Cli,
-) -> anyhow::Result<(CodePathBuf, StatePathBuf)> {
+) -> anyhow::Result<GenesisArtifacts> {
 	// Changes the working directory if a path is provided to ensure the build spec process runs in
 	// the correct context.
 	if let Some(path) = path {
 		std::env::set_current_dir(path)?;
 	}
-	let build_spec = BuildSpecCommand {
+	let mut build_spec = BuildSpecCommand {
 		id: Some(id),
 		genesis_code: true,
 		genesis_state: true,
+		chain_type: Some(ChainType::Live),
 		..Default::default()
 	}
 	.configure_build_spec(cli)
 	.await?;
 
-	let (genesis_code_file, genesis_state_file) = build_spec.build(cli)?;
-	Ok((
-		genesis_code_file.ok_or_else(|| anyhow::anyhow!("Failed to generate the genesis code."))?,
-		genesis_state_file
-			.ok_or_else(|| anyhow::anyhow!("Failed to generate the genesis state file."))?,
-	))
+	let mut genesis_artifacts = build_spec.clone().build(cli)?;
+	if let Some(api) = &deployment_config.api {
+		cli.info("Fetching collator keys...")?;
+		let keys = api.get_collator_keys(id).await?;
+		cli.info("Rebuilding chain spec with updated collator keys...")?;
+		build_spec
+			.update_chain_spec_with_keys(keys.collator_keys, &genesis_artifacts.chain_spec)?;
+		build_spec.enable_existing_plain_spec();
+		genesis_artifacts = build_spec.build(cli)?;
+		deployment_config.collator_file_id = Some(keys.collator_file_id);
+	}
+	Ok(genesis_artifacts)
 }
 
 // Prompt the user to input an address and return it formatted as `Id(address)`
 fn prompt_for_proxy_address(cli: &mut impl Cli) -> Result<String> {
+	cli.info(format!(
+		"Don't have a pure proxy?\n{}",
+		style("Create a proxy account using `pop call chain` and fund it with enough balance for the registration.").dim()
```

Comment:
```suggestion
		style("Create a proxy account using `pop call chain --pallet Proxy --function create_pure --args "Any()" "0" "0" --url ws://localhost:62785/ --use-wallet` and fund it with enough balance for the registration.").dim()
```

Can we change it to direct single command with arguments so user can simply copy paste? We should use common default values for the command. 

At this stage, we already have the information of the selected RPC URL right? If so, we should provide it to the `--url` argument to avoid user retyping it again when they call `pop call chain`

Reply:
Better messaging in: https://github.com/r0gue-io/pop-cli/pull/459/commits/a402e3e59ba6ac58ed10707446f3b8efe45f9231